activerecord/lib/arel/visitors/to_sql.rb

```diff
@@ -333,7 +333,7 @@ def visit_Arel_Nodes_Grouping(o, collector)
         def visit_Arel_Nodes_HomogeneousIn(o, collector)
           collector.preparable = false
 
-          collector << quote_table_name(o.table_name) << "." << quote_column_name(o.column_name)
+          visit o.left, collector
```

Comment:
Can you write a failing test for the behavior you're changing?

I'm not quite following the code example or the changes in the linked PR. Generally Arel is considered a private API so understanding what's broken will inform us whether we want to accept a change to this internal implementation.